modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_serializer.py

```diff
@@ -252,7 +277,7 @@ def serialize_literal(self, literal):
                 "type_scale": -2147483648,
                 "type_precision": len(literal.val),
             }
-        if type(literal.val) in (int, np.int8, np.int16, np.int32, np.int64):
+        if type(literal.val) in type(self)._INT_OPTS.keys():
```

Comment:
```suggestion
        if type(literal.val) in self._INT_OPTS.keys():
```
?

Reply:
No, it's insignificant, but if there are a lot of such pieces of code, it might have some performance impact.
This method performs computation to just print the warning. It could be useful for developers, but not the end users. End users hate warnings and file bug reports on any warning. If the warning is filtered out, the computation is still performed to do nothing. It seems like a waste of resources.

In my opinion, this code is useful for debugging purposes only. If the `assert` seems confusing, I could replace it with `if __debug__` statement.

Reply:
> This method performs computation to just print the warning. It could be useful for developers, but not the end users.

I don't know, this kind of conversion may affect the computation result significantly:
- overflow threshold for uint<64 is now different, for example, you won't get 255 + 1 == 0 for uint8
- uint64 doesn't support values greater than int64.max

I, as a user, want to be advised when something unexpected happens with my data, and uint->int conversion is one of these unexpected things.

Reply:
But you have this option - just do not disable assertions. And you have an option to disable unnecessary computation either.

Anyway, I do not insist on using asserts for such purpose. I understand it looks ugly. Probably, a better way would be to have some logging capabilities with condition callable. I.e., the logger prints the message if the callable returns true and do not calls it at all if the message is disabled.